x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/index/IndexResolver.java

```diff
@@ -163,16 +168,24 @@ public boolean equals(Object obj) {
     private final String clusterName;
     private final DataTypeRegistry typeRegistry;
 
-    public IndexResolver(Client client, String clusterName, DataTypeRegistry typeRegistry) {
+    private final RemoteClusterResolver remoteClusters;
```

Comment:
The name of the variable is misleading. Also, why not storing the clusters themselves in the first place, instead of the resolver?

Reply:
The cluster list is not static since remotes can go offline or be added/removed.